machine/v1alpha1/types_openstack.go

```diff
@@ -0,0 +1,358 @@
+	CIDR            string `json:"cidr,omitempty"`
+	IPv6AddressMode string `json:"ipv6AddressMode,omitempty"`
+	IPv6RAMode      string `json:"ipv6RaMode,omitempty"`
+	ID              string `json:"id,omitempty"`
+	SubnetPoolID    string `json:"subnetpoolId,omitempty"`
+	Limit           int    `json:"limit,omitempty"`
+	Marker          string `json:"marker,omitempty"`
+	SortKey         string `json:"sortKey,omitempty"`
+	SortDir         string `json:"sortDir,omitempty"`
+	Tags            string `json:"tags,omitempty"`
+	TagsAny         string `json:"tagsAny,omitempty"`
+	NotTags         string `json:"notTags,omitempty"`
+	NotTagsAny      string `json:"notTagsAny,omitempty"`
+}
+
+type PortOpts struct {
+	NetworkID           string        `json:"networkID" required:"true"`
+	NameSuffix          string        `json:"nameSuffix" required:"true"`
+	Description         string        `json:"description,omitempty"`
+	AdminStateUp        *bool         `json:"adminStateUp,omitempty"`
+	MACAddress          string        `json:"macAddress,omitempty"`
+	FixedIPs            []FixedIPs    `json:"fixedIPs,omitempty"`
+	TenantID            string        `json:"tenantID,omitempty"`
+	ProjectID           string        `json:"projectID,omitempty"`
+	SecurityGroups      *[]string     `json:"securityGroups,omitempty"`
+	AllowedAddressPairs []AddressPair `json:"allowedAddressPairs,omitempty"`
+	Tags                []string      `json:"tags,omitempty"`
+
+	// The ID of the host where the port is allocated
+	HostID string `json:"hostID,omitempty"`
+
+	// The virtual network interface card (vNIC) type that is bound to the
+	// neutron port.
+	VNICType string `json:"vnicType,omitempty"`
+
+	// A dictionary that enables the application running on the specified
+	// host to pass and receive virtual network interface (VIF) port-specific
+	// information to the plug-in.
+	Profile map[string]string `json:"profile,omitempty"`
+
+	// enable or disable security on a given port
+	// incompatible with securityGroups and allowedAddressPairs
+	PortSecurity *bool `json:"portSecurity,omitempty"`
+
+	// Enables and disables trunk at port level. If not provided, openStackMachine.Spec.Trunk is inherited.
+	Trunk *bool `json:"trunk,omitempty"`
+}
+
+type AddressPair struct {
+	IPAddress  string `json:"ipAddress,omitempty"`
+	MACAddress string `json:"macAddress,omitempty"`
+}
+
+type FixedIPs struct {
+	SubnetID  string `json:"subnetID"`
+	IPAddress string `json:"ipAddress,omitempty"`
+}
+
+type RootVolume struct {
+	SourceType string `json:"sourceType,omitempty"`
+	SourceUUID string `json:"sourceUUID,omitempty"`
+	DeviceType string `json:"deviceType"`
+	VolumeType string `json:"volumeType,omitempty"`
+	Size       int    `json:"diskSize,omitempty"`
+	Zone       string `json:"availabilityZone,omitempty"`
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// OpenstackClusterProviderSpec is the providerSpec for OpenStack in the cluster object
```

Comment:
What cluster object? We have a cluster object?

Reply:
I still don't understand this

Reply:
Me neither.

Reply:
This used to be referenced in legacy CAPO by both the [NetworkService controller](https://github.com/openshift/cluster-api-provider-openstack/blob/c9e85d4c3ee1a478c6f41d345b597f1e940280be/pkg/cloud/openstack/clients/networkservice.go#L51-L52) and the [SecGroupService controller](https://github.com/openshift/cluster-api-provider-openstack/blob/c9e85d4c3ee1a478c6f41d345b597f1e940280be/pkg/cloud/openstack/clients/secgroupservice.go#L70-L71). However, these were dead code (we didn't run those controllers), so we removed them in MAPO.

It's used in machineService but [we never actually read it from anywhere](https://github.com/openshift/cluster-api-provider-openstack/blob/c9e85d4c3ee1a478c6f41d345b597f1e940280be/pkg/cloud/openstack/machine/actuator.go#L223-L225).

It's not reference by any other type and it doesn't have a CRD associated with it, so I don't see how it's possible to even use it. I was trying to confine this to a lift & shift, but I wonder if we could just go ahead and delete this.

Reply:
If we are absolutely sure that its not being used and hasn't been used for a while then I think that makes sense